docs/data/data-grid/ai-assistant/ai-assistant.md

```diff
@@ -0,0 +1,163 @@
+  [apiRef, allowDataSampling],
+);
+```
+
+{{"demo": "AssistantWithDataSampling.js", "bg": "inline"}}
+
+## Using Server-side data
+
+An example of combining prompt control with the [Server-side data](/x/react-data-grid/server-side-data/)
+
+{{"demo": "AssistantWithDataSource.js", "bg": "inline"}}
+
+## Processing service integration
+
+Natural language prompts must be processed by a service to understand what kind of state changes must be applied to the Data Grid to match the user's request.
+You can use MUI's processing service or build it by yourself.
+
+### With MUI's service
+
+Data Grid provides all necessary pieces to make the service integration easy.
+
+1. Enable the AI Assistant feature by adding `enableAiAssistant` prop.
+   A new toolbar button will appear in the default [`<Toolbar />`](/x/react-data-grid/components/toolbar/).
+   This button opens `<AssistantPanel />` that can take user's prompts.
+2. Contact [sales@mui.com](mailto:sales@mui.com) to get an API key for our processing service.
+
+   :::error
+   Avoid exposing the API key to the client by using a proxy server that receives prompt processing requests, adds the `x-api-key` header, and passes the request further to the MUI's service.
+
+   This is an example of a [Fastify proxy](https://www.npmjs.com/package/@fastify/http-proxy) for the prompt requests.
+
+   ```ts
+   fastify.register(proxy, {
+     upstream: 'https://api.mui.com',
+     prefix: '/api/my-custom-path',
+     rewritePrefix: '/api/v1/datagrid/prompt',
+     replyOptions: {
+       rewriteRequestHeaders: (_, headers) => ({
+         ...headers,
+         'x-api-key': process.env.MUI_DATAGRID_API_KEY,
+       }),
+     },
+   });
+   ```
+
+   :::
+
+3. Provide `onPrompt()` callback to pass the user's prompts to the service.
+   Service's response will be used internally to make the necessary state updates.
+
+   :::success
+   You can implement `onPrompt()` with `unstable_gridDefaultPromptResolver()`.
+   It will add necessary headers and stringify the body in the right format for you.
+
+   In addition to this, it allows you to add an additional prompt context to achieve better processing results.
+
+   ```ts
+   const PROMPT_RESOLVER_PROXY_BASE_URL =
+     process.env.NODE_ENV === 'development'
+       ? 'http://localhost:3000'
+       : 'https://api.my-proxy.com';
+
+   function processPrompt(query: string, context: string) {
+     const additionalContext = `The rows represent: List of employees with their company, position and start date`;
+
+     return unstable_gridDefaultPromptResolver(
+       `${PROMPT_RESOLVER_PROXY_BASE_URL}/api/my-custom-path`,
+       query,
+       context,
+       additionalContext,
+     );
+   }
+   ```
+
+   :::
+
+### With custom service
+
+Use pieces of the AI Assistant feature to build your own prompt processing service.
+
+- `<GridAiAssistantPanel />` and `<GridPromptField />` components can be used to build custom UI.
```

Comment:
I think we should link to the composable components here instead:

```suggestion
- The [AI Assistant Panel](/x/react-data-grid/components/ai-assistant-panel/) and [Prompt Field](/x/react-data-grid/components/prompt-field/) components can be used to build custom UI.
```

Reply:
good call!